### PR TITLE
Workaround missing function debug name info in release artifacts

### PR DESCRIPTION
**Pull request by THenry14**
## Introduced changes

- Works around an issue, where there are no function debug name info in release artifacts. The issue will be fully fixed by https://github.com/foundry-rs/starknet-foundry/issues/2042

## Checklist

- [X] Linked relevant issue
- [X] Updated relevant documentation
- [X] Added relevant tests
- [X] Performed self-review of the code
- [X] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/tests/e2e/script/general.rs [hunk 1]
```diff
@@ -508,3 +508,41 @@ async fn test_state_file_rerun_failed_tx() {
         .unwrap();
     assert_tx_entry_success(invoke_tx_entry, "invoke");
 }
+
+#[tokio::test]
+async fn test_using_release_profile() {
```
Comment (ddoktorski) [inline crates/sncast/tests/e2e/script/general.rs:513]: It seems that this test also passes successfully without the changes from this PR (for example on the master branch). Could you check this?

File: crates/sncast/src/starknet_commands/script/run.rs [hunk 2]
```diff
@@ -454,9 +454,7 @@ fn inject_lib_artifact(
         .target_dir
         .clone()
         .unwrap_or_else(|| metadata.workspace.root.join("target"));
-    let sierra_path = &target_dir
-        .join(&metadata.current_profile)
-        .join(sierra_filename);
+    let sierra_path = &target_dir.join("dev").join(sierra_filename);
```
Comment (ddoktorski) [inline crates/sncast/src/starknet_commands/script/run.rs:457]: Let's also add `TODO #2042` here, so we won't forget to update it.

**Review by piotmag769 — COMMENTED** [review]
Maybe a better solution would be to tell users to override the release profile config https://docs.swmansion.com/scarb/docs/reference/profiles.html#overriding-built-in-profile-properties? It will affect contract compilation though cc @maciektr

**Comment by maciektr** [comment]
@piotmag769 I believe this is specifically done to allow a build with not replaced ids.

**Comment by piotmag769** [comment]
>I believe this is specifically done to allow a build with not replaced ids.

Doesn't calling build twice replace these artifacts anyways?

**Comment by piotmag769** [comment]
>I believe this is specifically done to allow a build with not replaced ids.

If that's the rationale then it's ok

File: crates/sncast/tests/e2e/script/general.rs (hunk 1 above)
Comment (THenry14) [reply crates/sncast/tests/e2e/script/general.rs:513]: it fails without the changes from this PR (clean master, only this test added)

```
---- e2e::script::general::test_using_release_profile stdout ----
thread 'e2e::script::general::test_using_release_profile' panicked at crates/sncast/tests/e2e/script/general.rs:540:32:

--- Expected
++++ actual:   stdout
   1      - ...
   2      - command: script run
   3      - status: success
        1 +    Compiling lib(map_script) map_script v0.1.0 (/private/var/folders/rd/q9ymr6vd6w150wm7g3f36hf00000gn/T/.tmp5uDNi6/Scarb.toml)
        2 +    Compiling starknet-contract(map_script) map_script v0.1.0 (/private/var/folders/rd/q9ymr6vd6w150wm7g3f36hf00000gn/T/.tmp5uDNi6/Scarb.toml)
        3 +     Finished release target(s) in 2 seconds

Exit status: 0
stderr:

command: script run
error: No such file or directory (os error 2)

```

File: crates/sncast/src/starknet_commands/script/run.rs (hunk 2 above)
Comment (THenry14) [reply crates/sncast/src/starknet_commands/script/run.rs:458]: done in newest commit

**Comment by THenry14** [comment]
> Maybe a better solution would be to tell users to override the release profile config (`sierra-replace-ids` property) https://docs.swmansion.com/scarb/docs/reference/profiles.html#overriding-built-in-profile-properties? It will affect contract compilation though cc @maciektr

Apart from what Maciek said, this should not be considered a proper fix, rather a workaround. We wait for `executable attributes` in scarb to properly resolve this issue.

**Review by piotmag769 — COMMENTED** [review]
Can we add a test for a case when dev profile is overriden like this in Scarb.toml:
```
[profile.dev.cairo]
sierra-replace-ids = false
```
and see what the error message is?

**Comment by THenry14** [comment]
> Can we add a test for a case when dev profile is overriden like this in Scarb.toml:
> 
> ```
> [profile.dev.cairo]
> sierra-replace-ids = false
> ```
> 
> and see what the error message is?

when I do this for dev I get the error with missing function debug name (Function with suffix `script::main` to run not found)

**Pull request merged**